flexmeasures/data/tests/test_train_predict_pipeline.py

```diff
@@ -203,6 +203,31 @@ def test_train_predict_pipeline(
             source
         ), "string representation of the Forecaster (DataSource) should mention the used model"
 
+        assert isinstance(pipeline_returns, list), "pipeline should return a list"
```

Comment:
This is nice. One more trivial check would be to check for a non-empty list. This is important, because the other `assert` statements are inside a for-loop over this list. So without this extra check, the test would not catch a regression causing `pipeline_returns` to become an empty list.
```suggestion
        assert isinstance(pipeline_returns, list) and len(pipeline_returns) > 0, "pipeline should return a non-empty list"
```